srml/support/procedural/tools/derive/src/lib.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2017-2018 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+// tag::description[]
+//! Use to derive parsing for parsing struct.
+// end::description[]
+
+
+#![recursion_limit = "128"]
+
+#[macro_use]
+extern crate syn;
+
+#[macro_use]
+extern crate quote;
+
+extern crate proc_macro;
+extern crate proc_macro2;
+
+use proc_macro::TokenStream;
+
+
+pub(crate) fn fields_idents(
+	fields: impl Iterator<Item = syn::Field>,
+) -> impl Iterator<Item = proc_macro2::TokenStream> {
+	fields.enumerate().map(|(ix, field)| {
+		field.ident.clone().map(|i|quote!{#i}).unwrap_or_else(|| {
+			let f_ix: syn::Ident = syn::parse_str(&format!("f_{}", ix))
```

Comment:
`Ident::new("name", Span::call_site())` is much cleaner and does not require an `unwrap`.